contracts/NodeStorage.sol

```diff
@@ -98,7 +101,32 @@ contract NodeStorage {
         }
     }
 
-    function calculateKey(string memory _enodeId, string memory _ip, uint16 _port) public pure returns(uint256) {
-        return uint256(keccak256(abi.encodePacked(_enodeId, _ip, _port)));
+    function setValidateEnodeIdOnly (bool activate) public onlyLatestVersion returns (bool) {
+        if (hostValidation == activate) {
```

Comment:
we flipped the name but not the boolean so this is confusing 
setValidateEnodeIdOnly(bool onlyUseEnodeId)